### PR TITLE
Meckstss code review1

### PR DESCRIPTION
**Pull request by SteveMeckstroth**

### CODE REVIEW

**Comment by SteveMeckstroth** [comment]
Overall I'm impressed with the working program you have working with no compile errors.  Good job!  This program was available well ahead of schedule to begin the code review.  I was a bit challenged to follow the program due to the minimal javadoc, but it closely followed the lectures so I was able to figure it out.

**Comment by SteveMeckstroth** [comment]
I also wanted to point out that I was impressed with your team's knowledge of CSS.   I like the calendar design, and I may look to do something like that in a future project.   I also learned the value of javadoc on a project where I have had no involvement prior to this code review.   Finally, I learned about how to create an error page because I had to find something new to implement :)

File: src/main/java/com/moodybluez/enterprise/MoodyBluezController.java [hunk 1]
```diff
@@ -1,17 +1,52 @@
...
     @RequestMapping("/")
     public String index() {
         return "index";
     }
+
+    /**
+     * Returns the json representation of a Mood by id
+     * @param id integer id of a mood
+     * @return Mood in JSON format
+     * @author Stephen Meckstroth
```
Comment (discospiff) [inline src/main/java/com/moodybluez/enterprise/MoodyBluezController.java:29]: param and return are good, but the author tag is somewhat deprecated.

That was handy back in the days before good version control (believe it or not, there were days like that.  Now, though, we can see change history and the person who did it in version control, so no need for the author tag.

File: src/main/java/com/moodybluez/enterprise/dao/MoodDAO.java [hunk 2]
```diff
@@ -5,16 +5,29 @@
...
     @Override
     public boolean createEntry(Mood mood) {
-        moods.put(mood.getMoodID(), mood);
-        if (moods.containsKey(mood.getMoodID())) return true;
-        return false;
+        try {
+            moods.put(mood.getMoodID(), mood);
+            return true;
+        } catch( Exception e) {
+            return false;
```
Comment (discospiff) [inline src/main/java/com/moodybluez/enterprise/dao/MoodDAO.java:29]: Don't just catch an exception and return false, as you're throwing away a lot of good information here.  Better to rethrow the exception than to bury it and return false.  Log the exception, at the very least.

If an exception were to occur here, you'd have no idea why.

File: src/main/java/com/moodybluez/enterprise/MoodyBluezController.java [hunk 3]
```diff
@@ -29,4 +34,19 @@ public String index() {
...
+     */
+    @RequestMapping("/error")
+    public String handleError(HttpServletRequest request) {
+        return "error";
+    }
+
+    @Override
+    public String getErrorPath() {
+        return null;
```
Comment (discospiff) [inline src/main/java/com/moodybluez/enterprise/MoodyBluezController.java:50]: return null?  That doesn't look like a proper implementation.

**Pull request closed without merging**